assets/src/edit-story/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -101,17 +104,23 @@ const getImageResource = async (file) => {
 const getVideoResource = async (file) => {
   const fileName = getFileName(file);
   const mimeType = file.type;
+
   const reader = await createFileReader(file);
-  const src = window.URL.createObjectURL(
-    new window.Blob([reader.result], { type: mimeType })
-  );
+
+  const src = createBlob(new Blob([reader.result], { type: mimeType }));
+
+  const videoEl = document.createElement('video');
+  const canPlayVideo = '' !== videoEl.canPlayType(mimeType);
```

Comment:
Very cool!

Reply:
`canPlayType` only supports 1 parameter, the mime type.

Possible return values are `''` (empty string), `maybe` and `probably`.

An empty string means the file definitely can't be played on the current device. But the other two values are no guarantees either way. But that's OK, as those values are not being relied upon.

Reply:
> canPlayType only supports 1 parameter, the mime type.

You can pass codex in that one param. [See](https://www.w3schools.com/tags/av_met_canplaytype.asp).

Doing a little research, it is not easy to read codex from a file...

Reply:
I'm a bit skeptic about this not being documented on MDN...

And yeah we don't know the codec here.

`canPlayType` should be good enough here.

Reply:
The MDN docs are weird here. Look [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement) it does document codecs, but not on this [page](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canPlayType).

> Given a string specifying a MIME media type (potentially with the codecs parameter included)